expression/builtin_arithmetic_vec.go

```diff
@@ -429,13 +429,128 @@ func (b *builtinArithmeticDivideRealSig) vecEvalReal(input *chunk.Chunk, result
 }
 
 func (b *builtinArithmeticIntDivideIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinArithmeticIntDivideIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
-}
+	n := input.NumRows()
+	lh, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(lh)
+
+	if err := b.args[0].VecEvalInt(b.ctx, input, lh); err != nil {
+		return err
+	}
+
+	if err := b.args[1].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	result.MergeNulls(lh)
+
+	rh := result
+	lhi64s := lh.Int64s()
+	rhi64s := rh.Int64s()
 
+	resulti64s := result.Int64s()
+
+	isLHSUnsigned := mysql.HasUnsignedFlag(b.args[0].GetType().Flag)
+	isRHSUnsigned := mysql.HasUnsignedFlag(b.args[1].GetType().Flag)
+	fmt.Println(rhi64s)
+	switch {
+	case isLHSUnsigned && isRHSUnsigned:
+		err = b.divideUU(rh, result, lhi64s, rhi64s, resulti64s)
+	case isLHSUnsigned && !isRHSUnsigned:
+		err = b.divideUI(rh, result, lhi64s, rhi64s, resulti64s)
+	case !isLHSUnsigned && isRHSUnsigned:
+		err = b.divideIU(rh, result, lhi64s, rhi64s, resulti64s)
+	case !isLHSUnsigned && !isRHSUnsigned:
+		err = b.divideII(rh, result, lhi64s, rhi64s, resulti64s)
+	}
+	return err
+}
+func (b *builtinArithmeticIntDivideIntSig) divideUU(rhs, result *chunk.Column, lhi64s, rhi64s, resulti64s []int64) error {
+	for i := 0; i < len(lhi64s); i++ {
+		if !rhs.IsNull(i) && rhi64s[i] == 0 {
```

Comment:
Can this code block be moved outside of `divideXY`? If so, we can check `if divisor[i] == 0{ continue }` in line 483, and remove the input arg `result`